src/Group.cpp

```diff
@@ -0,0 +1,261 @@
+ * being developed.
+*/
+const RString INI_FILE = "Pack.ini";
+const int INI_VERSION = 1;
+
+Group::Group() {
+    m_sDisplayTitle = "";
+    m_sSortTitle = "";
+    m_sPath = "";
+    m_sGroupName = "";
+    m_sTranslitTitle = "";
+    m_sSeries = "";
+    m_fSyncOffset = 0;
+    m_bHasPackIni = false;
+    m_iYearReleased = 0;
+    m_sBannerPath = "";
+}
+
+Group::~Group() {
+    SONGMAN->GetGroupGroupMap().clear();
+}
+
+Group::Group(const RString& sDir, const RString& sGroupDirName) {
+    RString sPackIniPath = sDir + sGroupDirName + "/" + INI_FILE;
+    RString sDisplayTitle = sGroupDirName;
+    RString sSortTitle = sGroupDirName;
+    RString sTranslitTitle = "";
+    RString sSeries = "";
+    RString sBannerPath = "";
+    float fOffset = PREFSMAN->m_fMachineSyncBias;
+
+    if (FILEMAN->DoesFileExist(sPackIniPath)) {
+        IniFile ini;
+        ini.ReadFile(sPackIniPath);
+        ini.GetValue("Group", "DisplayTitle", sDisplayTitle);
+        Trim(sDisplayTitle);
+        if (sDisplayTitle.empty()) {
+            sDisplayTitle = Basename(sGroupDirName);
+        }
+        ini.GetValue("Group", "Banner", sBannerPath);
+        ini.GetValue("Group", "SortTitle", sSortTitle);
+        Trim(sSortTitle);
+        if (sSortTitle.empty()) {
+            sSortTitle = sDisplayTitle;
+        }
+        ini.GetValue("Group", "TranslitTitle", sTranslitTitle);
+        Trim(sTranslitTitle);
+        if (sTranslitTitle.empty()) {
+            sTranslitTitle = sDisplayTitle;
+        }
+        ini.GetValue("Group", "Series", sSeries);
+        RString sValue = "";
+        
+        ini.GetValue("Group", "SyncOffset", sValue);
+        Trim(sValue);
+        if (!sValue.empty()) {
+            if (sValue.CompareNoCase("null") == 0) {
+                fOffset = 0.0f;
+            } else if (sValue.CompareNoCase("itg") == 0) {
+                fOffset = -0.009f;
+            } else {
+                fOffset = StringToFloat(sValue);
+            }
+        }
+        ini.GetValue("Group", "Year", m_iYearReleased);
+        
+        RString sVersion = "";
+        ini.GetValue("Group", "Version", sVersion);
+        Trim(sVersion);
+        if (!sVersion.empty()) {
+            m_iVersion = StringToInt(sVersion);
+        } else {
+            // If the version is not set, set it to the current version
+            // and write it to the file.
+            m_iVersion = INI_VERSION;
+            ini.SetValue("Group", "Version", INI_VERSION);
+            if( !ini.WriteFile(sPackIniPath)) {
+                RString sError = ssprintf( "Error writing pack.ini ", sPackIniPath.c_str(), ini.GetError().c_str() );
+                Dialog::OK( sError );
+            }
+        }
+
+    } else {
+        m_bHasPackIni = false;
+        m_iVersion = INI_VERSION;
+        fOffset = PREFSMAN->m_fMachineSyncBias;
+    }
+
+    
+	// Look for a group banner in this group folder
+	std::vector<RString> arrayGroupBanners;
```

Comment:
I think we shouldn't do any of this here. I think here we should just set sBannerPath if not empty. Where the actual banner gets loaded is where we can check if it's set for in pack.ini and override the defaults. The getters for pack.ini should be whats actually specified in the file and not just implicitly computed. It becomes pretty opaque otherwise.

We should still make sure that wherever we get the banner path in the theme for a group (like SSM) should return the path that's actually used (i.e. specified in pack.ini if set, otherwise whatever fallback image the game picks).

Reply:
Restored previous behavior prior to PR